leshan-server-cluster/src/main/java/org/eclipse/leshan/server/cluster/RedisRegistrationStore.java

```diff
@@ -482,16 +495,26 @@ public void add(org.eclipse.californium.core.observe.Observation obs) {
                 String registrationId = ObserveUtil.extractRegistrationId(obs);
                 if (!j.exists(toRegIdKey(registrationId)))
                     throw new IllegalStateException("no registration for this Id");
-
-                byte[] previousValue = j.getSet(toKey(OBS_TKN, obs.getRequest().getToken()), serializeObs(obs));
+                byte[] key = toKey(OBS_TKN, obs.getRequest().getToken().getBytes());
+                byte[] serializeObs = serializeObs(obs);
+                byte[] previousValue = null;
+                if (ifAbsent) {
+                    previousValue = j.get(key);
+                    if (previousValue == null || previousValue.length > 0) {
```

Comment:
I still have concerns, if `previousValue.length > 0` is correct, I think it must be `previousValue.length == 0`.

Reply:
You're right ! I will fix that.